### ISSUE TITLE
View oriented N:N operations

### ISSUE THREAD

**Issue by ghost** [a33ik/UltimateWorkflowToolkit#35]
Hey Andrew,
first of all (I know, not the place, but hey) thanks for all the hard work so far. I'm enjoying your UWT every day!

In your releasenotes of 1.5.0.0 your first bullet was "I merged relationship-related (N-1 and N-N) operations with View-Oriented operations." which I interpreted as me being able to associate N:N by a view or fetchxml. Unfortunately that's not what it's doing/enabling me to.
Was I too ambitious/unlogical or am I just doing something wrong if I suspect that I can relate specific records to another but not arrays from a view or fetch?

Thanks a million and best regards!
Jeroen

**Comment by AndrewButenko** [comment]
Hello Jeroen,
I believe description was a bit misleading but the intention was following:
I have a set of operations that I called "Bulk Operations" like "Distribute Workflow" or "Distribute Action".

Before 1.5.0.0 there were separated operations for "Bulk Operations" across 1-N/N-N relationships, generic FetchXml, Public or Private view.

In 1.5.0.0 I merged those actions into one and customizer during the design of the workflow can choose the set of records populating either Parent Record and 1-N/N-N relationship OR Public View OR Private View OR generic FetchXml to do the bulk operation (either Distribute Workflow or Distribute Action or Update Records).

Feel free to update the thread if you have questions.

Andrew

**Comment by ghost** [comment]
Hi Andrew,
Thanks for your quick reply. I didn't really believe I actually found a flaw in your work, hád to be an interpretation error ;)

Nah, questions... eeerm, well not really. I'm still searching for a way to create N:N relationships between the caller entity record and records from a (complex) fetchxml, but I'm afraid I have to continue my quest.

Still, happy with your product every single (working) day!
Thanks again!

**Comment by AndrewButenko** [comment]
Hey,
That's doable using my toolkit. Direction is Action (that will do the association using step from the toolkit) and Distribute Action across result of "FetchXml" query. Something similar is described here - http://butenko.pro/2018/04/26/cloning-record-with-related-records-with-no-code-using-uwt/

Andrew

**Comment by ghost** [comment]
Wow! Great thing to work with, never expected a route like this to work.
Could you please point me out in the right direction for my specific quest?

Goal: 
Running a workflow from entity A to associate (n:n) A to several records in entity B that have a specific characteristic.

My thoughts regarding to your hand out:
Create Action on entity B with a Entity Reference to A and your Associate Entities step that associates URL from A and B based on their relationship name.
Next I create a workflow on A with your Distribute Action pointing to the earlier created action, distributed to my fetchxml querying for the characteristic.

It looks promising, but I know I'm missing something here, passionately fighting my lack of experience with the Action category.

**Comment by AndrewButenko** [comment]
Jeroen,

You described exactly what you should do to accomplish your task. Feel free to update the thread if you have issues.

Andrew

**Comment by ghost** [comment]
Really?? Whaha! Then I should inspect my work more carefully (didn't get it to work yet, but I will).

I'll see what happens if I follow my own walkthrough again and let you know the outcome.

Thanks!

Jeroen

**Comment by ghost** [comment]
Hey Andrew,
In my particular case, is there also need to build a Request? I can't really find any need for it, but I also haven't got things to work. The only difference I can find is the Request and the referral to it in the Associate Entities properties.

(few hours later ;))
Awr, never mind that question. The good old "why don't you just try it out" did it's trick again.

Still not the result (none) that I wanted, but I think I'll manage now that I got to understand how things cling together.

Jeroen

**Comment by AndrewButenko** [comment]
Jeroen,

You will have to build the request to pass a reference to your current record inside.

Andrew

**Comment by ghost** [comment]
Exactly! And that was the (critical) step I was overlooking in the first place. With that referral in place and understanding the different fields of Distribute Action, I've got it running as smooth as a bowling green.

Thanks for getting me up to speed, Andrew! Thanks for hanging on with me and thanks for your gorgeous solutions!

Best regards,
Jeroen
